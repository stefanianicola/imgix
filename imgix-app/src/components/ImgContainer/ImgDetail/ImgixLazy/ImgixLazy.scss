img {
    &.imgSelected {
        width: 100%;
    }
}
