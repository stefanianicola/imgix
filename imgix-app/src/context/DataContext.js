import React, { createContext, useState, useEffect } from 'react';
import axios from 'axios';
import Params from '../Data/Params';
export const DataContext = createContext();


const DataContextProvider = (props) => {
    const [data, setData] = useState([]);
    const [flag, setFlag] = useState(false);
    const [btn, setBtn] = useState([]);
    const [nameButtons, setNameButtons] = useState([]);
    const [loading, setLoading] = useState(false);
    const [params, setParams] = useState([]);
    const [url, setUrl] = useState("");
    const [hideMenu, setHideMenu] = useState(false);

    const urlImg = "https://assets.imgix.net/unsplash/";

    //consume photos API
    useEffect(() => {
        setLoading(true);
        try {
            axios("https://storage.googleapis.com/nanlabs-engineering-technical-interviews/imgix-samples-list.json")
                .then((res) => {
                    setData(res.data);
                })
        } catch (error) {
            console.log(error);
        }
        setLoading(false);
    }, [])

    //Pass data from the params to the state of btn so that the names appear in the buttons
    useEffect(() => {
        setNameButtons(Params);
    }, [])


    //Pass the parameters of the button that you click
    const addControl = (v) => {
        let parametros = [...Params]
        parametros.forEach((param) => {
            if (v === param.alias) {
                setBtn(param)
            }
        })
        setFlag(true);
    }

    // filter the data of the params to be able to take a single value per parameter
    // and build the url with all the effects we want in the photos
    const addParamUrl = (alias, value) => {
        let aux = [...params];
        let isParam = false;

        aux.forEach((a) => {
            if (a.alias === alias) {
                isParam = true;
                a.value = value;
                setParams(aux);
            }
        })
        !isParam &&
            setParams([...aux, { alias, value }]);
        convertUrl();

    }

    //Convert the array of objects into a string to use as the url of the image
    const convertUrl = () => {
        let stringP = "";
        for (let i = 0; i < params.length; i++) {
            stringP = stringP + `${params[i].alias}=${params[i].value}&`

        }
        setUrl(stringP);

    }

    //reset url with change img
    const resetUrl = () => {
        setUrl("");
        setParams([]);
        setFlag(false);
    }

    //delete the last parameter added
    const removeItem = () => {
        params.pop();
        convertUrl();
        setFlag(false);
    }

    return (
        <DataContext.Provider
            value={{
                data,
                setData,
                addControl,
                nameButtons,
                flag,
                btn,
                loading,
                params,
                setParams,
                addParamUrl,
                url,
                removeItem,
                hideMenu,
                setHideMenu,
                resetUrl,
                urlImg
            }}>
            {props.children}
        </DataContext.Provider>
    )
}

export default DataContextProvider;
