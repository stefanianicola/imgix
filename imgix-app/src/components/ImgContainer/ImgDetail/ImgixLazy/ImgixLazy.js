import React, { useContext } from 'react';
import './ImgixLazy.scss';
import { useParams } from 'react-router-dom';
import 'bootstrap/dist/css/bootstrap.min.css';
import { DataContext } from '../../../../context/DataContext';
import Imgix from "react-imgix";
import 'font-awesome/css/font-awesome.min.css';

const ImgixLazy = () => {
    const { urlImg, url } = useContext(DataContext);
    const { name } = useParams();

    return (
        <Imgix className='imgSelected'
            alt=""
            src={`${urlImg}${name}?${url}`}
            sizes="calc(10% - 10px)"
        />
    )
}
export default ImgixLazy;