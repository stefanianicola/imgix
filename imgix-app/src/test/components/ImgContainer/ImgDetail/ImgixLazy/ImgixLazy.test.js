import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import ImgixLazy from '../../../../../components/ImgContainer/ImgDetail/ImgixLazy/ImgixLazy'
import { DataContext } from '../../../../../context/DataContext';

describe('ImgixLazy component', () => {
    const urlImg = "https://assets.imgix.net/unsplash/";
    const url = "exp=11&";
    const wrapper = mount(
        <Router>
            <DataContext.Provider value={{ urlImg, url }} >
                <ImgixLazy />
            </DataContext.Provider>
        </Router>
    );


    it('should be create the component', () => {
        expect(wrapper).toMatchSnapshot();
    });

})