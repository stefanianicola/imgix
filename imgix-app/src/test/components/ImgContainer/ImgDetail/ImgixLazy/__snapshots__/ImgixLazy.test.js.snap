// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImgixLazy component should be create the component 1`] = `
<img
  className="imgSelected"
  sizes="calc(10% - 10px)"
  src="https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1"
  srcSet="https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=100 100w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=116 116w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=135 135w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=156 156w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=181 181w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=210 210w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=244 244w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=283 283w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=328 328w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=380 380w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=441 441w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=512 512w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=594 594w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=689 689w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=799 799w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=927 927w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=1075 1075w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=1247 1247w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=1446 1446w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=1678 1678w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=1946 1946w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=2257 2257w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=2619 2619w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=3038 3038w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=3524 3524w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=4087 4087w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=4741 4741w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=5500 5500w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=6380 6380w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=7401 7401w,
https://assets.imgix.net/unsplash/undefined?exp=11&=undefined&auto=format&ixlib=react-9.1.1&w=8192 8192w"
/>
`;
